### PR TITLE
refactor: use ConfiguratorBundle interface for config generate

### PR DESCRIPTION
**Pull request by rsmitty**
This PR overhauls osctl's config generate command to make use of the new
ConfigBundle implementation of the ConfiguratorBundle interface

Signed-off-by: Spencer Smith <robertspencersmith@gmail.com>

### CODE REVIEW

File: internal/app/machined/internal/phase/platform/platform.go [hunk 1]
```diff
@@ -56,9 +56,6 @@ func (task *Platform) runtime(r runtime.Runtime) (err error) {
 	}
 
 	if r.Platform().Mode() == runtime.Container {
-		// TODO: add ::1 back once I figure out why bootkube barfs
```
Comment (rsmitty) [inline internal/app/machined/internal/phase/platform/platform.go]: We handle injecting this in cluster.go now. Was causing duplicates in machine sans.

File: cmd/osctl/cmd/cluster.go [hunk 2]
```diff
@@ -139,24 +134,25 @@ func create(ctx context.Context) (err error) {
...
 		if forceEndpoint != "" {
-			genOptions = append(genOptions, generate.WithEndpointList([]string{forceEndpoint}))
+			endpointList = append(endpointList, forceEndpoint)
 			provisionOptions = append(provisionOptions, provision.WithEndpoint(forceEndpoint))
 		} else if forceInitNodeAsEndpoint {
-			genOptions = append(genOptions, generate.WithEndpointList([]string{ips[0].String()}))
+			endpointList = append(endpointList, ips[0].String())
 		}
 
+		endpointList = append(endpointList, "127.0.0.1")
```
Comment (andrewrynhard) [inline cmd/osctl/cmd/cluster.go]: nit: We should have a comment here prefixed with `NB:` to indicate that it is crucial to set this _last_: https://en.wikipedia.org/wiki/Nota_bene .

Comment (smira) [reply cmd/osctl/cmd/cluster.go]: I wonder if it should be more of `switch`-like logic here? As right now it appends to the list, while it was supposed to be replacing the list depending on the config? 

Once we do client LB, having 127.0.0.1 while it's not running current cluster might lead to fun results.

Comment (andrewrynhard) [reply cmd/osctl/cmd/cluster.go]: I _think_ we append only for the purposes of getting a valid cert SANs? @rsmitty is that right?

Comment (rsmitty) [reply cmd/osctl/cmd/cluster.go]: fixed in amend.

**Pull request merged**